plugins/out_azure_kusto/azure_kusto_ingest.c

```diff
@@ -0,0 +1,430 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2015-2022 The Fluent Bit Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <fluent-bit/flb_http_client.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_random.h>
+#include <fluent-bit/flb_time.h>
+#include <fluent-bit/flb_utils.h>
+
+#include <math.h>
+#include <mbedtls/base64.h>
+#include <msgpack.h>
+
+#include "azure_kusto_ingest.h"
+
+/* not really uuid but a random string in the form 00000000-0000-0000-0000-000000000000 */
+static char *generate_uuid()
+{
+    char *chars = "0123456789abcdef";
+    char *uuid;
+    int i;
+    uint64_t rand;
+
+    uuid = flb_malloc(36);
+    if (!uuid) {
+        flb_errno();
+        return NULL;
+    }
+
+    for (i = 0; i < 36; i++) {
+        if (i == 8 || i == 13 || i == 18 || i == 23) {
+            uuid[i] = '-';
+            continue;
+        }
+
+        if (flb_random_bytes((unsigned char *)&rand, sizeof(uint64_t))) {
+            rand = time(NULL);
+        }
+        uuid[i] = chars[rand % 16];
+    }
+
+    return uuid;
+}
+
+static char *base64_encode(flb_sds_t s, size_t len, size_t *out_len)
```

Comment:
Is there a particular reason to have this function instead of using `flb_base64_encode`? I think `flb_base64_encode` is based on an older version of the mbedtls base64 function because there were performance issues with the latest version.

Regardless of that I'd encourage you to use `flb_base64_encode` because the underlying implementation might be improved soon and using this will result in an undesired dependency and additional work.